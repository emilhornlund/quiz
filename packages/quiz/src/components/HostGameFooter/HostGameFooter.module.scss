@import '../../styles/helpers';

.main {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  background-color: rgba(255, 255, 255, 0.5);

  @include mobile {
    border-radius: $layout-gutter * 0.5;
    padding: $layout-gutter * 0.5;
    width: 100%;
  }

  @include tablet {
    border-radius: $layout-gutter * 0.75;
    padding: $layout-gutter * 0.75;
    width: $tablet-width;
  }

  @include desktop {
    border-radius: $layout-gutter;
    padding: $layout-gutter;
    width: $desktop-width;
  }

  .questions, .gamePIN {
    display: flex;
    flex-direction: row;
    align-items: center;
    flex: 1;

    @include mobile {
      column-gap: $layout-gutter * 0.5;

      @include fontSize(16px);
    }

    @include tablet {
      column-gap: $layout-gutter * 0.75;

      @include fontSize(20px);
    }

    @include desktop {
      column-gap: $layout-gutter;

      @include fontSize(24px);
    }

    span {
      margin-top: -1%;
      font-weight: bold;
    }
  }

  .questions {
    justify-content: flex-start;
  }

  .gamePIN {
    justify-content: center;
  }

  .actions {
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    align-items: center;
    flex: 1;

    @include mobile {}
    @include tablet {}
    @include desktop {}
  }
}