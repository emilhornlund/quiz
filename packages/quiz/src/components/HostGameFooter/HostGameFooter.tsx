import React, { FC } from 'react'

import styles from './HostGameFooter.module.scss'

export interface HostGameFooterProps {
  gamePIN: string
  currentQuestion: number
  totalQuestions: number
}

const HostGameFooter: FC<HostGameFooterProps> = ({
  gamePIN,
  currentQuestion,
  totalQuestions,
}) => (
  <div className={styles.main}>
    <div className={styles.questions}>
      {currentQuestion} / {totalQuestions}
    </div>
    <div className={styles.gamePIN}>
      Game PIN: <span>{gamePIN}</span>
    </div>
    <div className={styles.actions} />
  </div>
)

export default HostGameFooter
