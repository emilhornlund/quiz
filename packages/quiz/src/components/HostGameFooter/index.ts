export type { HostGameFooterProps } from './HostGameFooter'
export { default } from './HostGameFooter'
