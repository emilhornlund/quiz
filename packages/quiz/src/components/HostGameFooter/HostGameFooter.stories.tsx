import type { Meta, StoryObj } from '@storybook/react'

import HostGameFooter from './HostGameFooter'

const meta = {
  component: HostGameFooter,
} satisfies Meta<typeof HostGameFooter>

export default meta
type Story = StoryObj<typeof meta>

export const Default = {
  args: {
    gamePIN: '123456',
    currentQuestion: 1,
    totalQuestions: 20,
  },
} satisfies Story
