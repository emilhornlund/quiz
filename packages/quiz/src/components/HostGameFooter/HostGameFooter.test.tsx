import { render } from '@testing-library/react'
import React from 'react'
import { describe, expect, it } from 'vitest'

import HostGameFooter from './HostGameFooter'

describe('HostGameFooter', () => {
  it('should render HostGameFooter with default props', () => {
    const { container } = render(
      <HostGameFooter
        gamePIN="123456"
        currentQuestion={1}
        totalQuestions={20}
      />,
    )

    expect(container).toMatchSnapshot()
  })
})
