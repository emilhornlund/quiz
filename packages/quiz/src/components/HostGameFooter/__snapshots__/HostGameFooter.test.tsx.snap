// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HostGameFooter > should render HostGameFooter with default props 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="questions"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-circle-question "
        data-icon="circle-question"
        data-prefix="fas"
        role="img"
        viewBox="0 0 512 512"
      >
        <path
          d="M256 512a256 256 0 1 0 0-512 256 256 0 1 0 0 512zm0-336c-17.7 0-32 14.3-32 32 0 13.3-10.7 24-24 24s-24-10.7-24-24c0-44.2 35.8-80 80-80s80 35.8 80 80c0 47.2-36 67.2-56 74.5l0 3.8c0 13.3-10.7 24-24 24s-24-10.7-24-24l0-8.1c0-20.5 14.8-35.2 30.1-40.2 6.4-2.1 13.2-5.5 18.2-10.3 4.3-4.2 7.7-10 7.7-19.6 0-17.7-14.3-32-32-32zM224 368a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"
          fill="currentColor"
        />
      </svg>
      <span>
        1
         / 
        20
      </span>
    </div>
    <div
      class="gamePIN"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-lock-open "
        data-icon="lock-open"
        data-prefix="fas"
        role="img"
        viewBox="0 0 576 512"
      >
        <path
          d="M384 96c0-35.3 28.7-64 64-64s64 28.7 64 64l0 32c0 17.7 14.3 32 32 32s32-14.3 32-32l0-32c0-70.7-57.3-128-128-128S320 25.3 320 96l0 64-160 0c-35.3 0-64 28.7-64 64l0 224c0 35.3 28.7 64 64 64l256 0c35.3 0 64-28.7 64-64l0-224c0-35.3-28.7-64-64-64l-32 0 0-64z"
          fill="currentColor"
        />
      </svg>
      <span>
        123456
      </span>
    </div>
    <div
      class="actions"
    />
  </div>
</div>
`;
